### PR TITLE
Memop bailout on negative length

### PR DESCRIPTION
**Pull request by Cellule**
Bailout on Memset/Memcopy if the length is <= 0. 
After the memop, we restore the induction variable to its expected value. 
If the memop didn't actually happen we shouldn't modify the induction variable.
This is unlikely to happen since we would normally bailout on no profile and not emit this memop.

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/chakracore/1275)

### CODE REVIEW

**Comment by Cellule** [comment]
@LouisLaf @rajatd @pleath @MikeHolman Could one of you review this please  ?

**Comment by MikeHolman** [comment]
lgtm

**Comment by satheeshravi** [comment]
:shipit:

File: test/Array/bug8159763.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+function test0() {
```
Comment (satheeshravi) [inline test/Array/bug8159763.js]: do you want to add to one of the existing memop test cases - instead of adding a file with bug number ?

Comment (Cellule) [inline test/Array/bug8159763.js]: Good idea, I check if I can get this done

Comment (Cellule) [inline test/Array/bug8159763.js]: Actually I can't because it need additional host flags to test properly.

**Pull request merged**